src/Microsoft.AspNet.Mvc.Core/ActionResults/ContentResult.cs

```diff
@@ -21,7 +22,9 @@ public class ContentResult : ActionResult
 
             if (!string.IsNullOrEmpty(ContentType))
             {
-                response.ContentType = ContentType;
+                var contentTypeHeader = new MediaTypeHeaderValue(ContentType);
+                contentTypeHeader.Encoding = ContentEncoding;
```

Comment:
ContentEncoding can be null if the user does not provide it...in this case should we default to UTF8 as that is what the HttpResponse object seems to be doing in the end? @Tratcher ...fyi:
https://github.com/aspnet/HttpAbstractions/blob/7be9cefc4c7671caba7d8fe538ba6a0ad308fdad/src/Microsoft.AspNet.Http/HttpResponseWritingExtensions.cs#L25